contracts/retirement/KlimaRetirementAggregator.sol

```diff
@@ -0,0 +1,388 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+import "../interfaces/IUniswapV2Router02.sol";
+import "../interfaces/IwsKLIMA.sol";
+import "../interfaces/IRetireBridgeCommon.sol";
+import "../interfaces/IRetireMossCarbon.sol";
+import "../interfaces/IRetireToucanCarbon.sol";
+
+/**
+ * @title KlimaRetirementAggregator
+ * @author KlimaDAO
+ *
+ * @notice This is the master aggregator contract for the Klima retirement utility.
+ *
+ * This allows a user to provide a source token and an approved carbon pool token to retire.
+ * If the source is different than the pool, it will attempt to swap to that pool then retire.
+ */
+contract KlimaRetirementAggregator is
+    Initializable,
+    ContextUpgradeable,
+    OwnableUpgradeable
+{
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    function initialize() public initializer {
+        __Ownable_init();
+        __Context_init();
+    }
+
+    /** === State Variables and Mappings === */
```

Comment:
Since you have started using upgradeable contracts, I'd recommend maintaining storage variables in separate contracts to make upgrades easier to maintain/test. Check the Ribbon contracts as an example:
https://github.com/ribbon-finance/ribbon-v2/blob/master/contracts/vaults/BaseVaults/RibbonThetaVault.sol
https://github.com/ribbon-finance/ribbon-v2/blob/master/contracts/storage/RibbonThetaVaultStorage.sol